### ISSUE TITLE
how does training parser for custom semantics works?

### ISSUE THREAD

**Issue by curiousgeek0** [explosion/spaCy#4478]
I am referring to [this part](https://spacy.io/usage/training#intent-parser) of documentation. 

Does it takes into consideration the meaning of words while labeling dependencies. 

I have multiple entities of same type in a sentence and I want to check their dependencies on some words to distinguish those entities.

For ex: Workshop starts at 9 am and ends at 10 pm.
Here 9 am , 10 pm are time entities.
How do I get to recognize 9 am as start_time and 10 pm as end_time.
How do I get my model to work even when starts is replaced with starting or strt (misspelled form) or begin.
How do i get my model to label correctly even if the positions are reversed, '
" Workshop ends at 10 am and starts at 9 am " 

## Your Environment

* Operating System:
* Python Version Used:
* spaCy Version Used:
* Environment Information:

**Comment by ines** [comment]
For details on how the transition-based parsing algorithm works, you might find this video helpful: https://www.youtube.com/watch?v=sqDHBH9IjRU It discusses the NER model, but the underlying idea is the same.

From what you describe, training a custom parser might not be the best solution here. It depends on the use case, but in your examples, all the information is already there – you just need to extract it. Maybe start with implementing a few information extraction rules that use the parser and tagger: https://spacy.io/usage/rule-based-matching#models-rules-pos-dep This also lets you handle the reversed positions, because the dependency attachment will still be the same. If the pre-trained parser is inaccurate on the data, you can always fine-tune it to improve it.
